git-branchless-lib/src/core/check_out.rs

```diff
@@ -71,20 +71,36 @@ pub fn check_out_commit(
         render_smartlog,
     } = options;
 
-    let target = match target {
-        None => None,
+    let (target, oid) = match target {
+        None => (None, None),
         Some(CheckoutTarget::Reference(reference_name)) => {
             let categorized_target = CategorizedReferenceName::new(&reference_name);
-            Some(categorized_target.remove_prefix()?)
+            (Some(categorized_target.remove_prefix()?), None)
         }
-        Some(CheckoutTarget::Oid(oid)) => Some(oid.to_string()),
-        Some(CheckoutTarget::Unknown(target)) => Some(target),
+        Some(CheckoutTarget::Oid(oid)) => (Some(oid.to_string()), Some(oid)),
+        Some(CheckoutTarget::Unknown(target)) => (Some(target), None),
     };
 
     if get_undo_create_snapshots(repo)? {
         create_snapshot(effects, git_run_info, repo, event_log_db, event_tx_id)?;
     }
 
+    let target = if let Some(oid) = oid {
```

Comment:
For consistency with the rest of the codebase, use `match oid { Some(oid) => { ... }, None => { ... } }`.